stonesoup/types/tests/test_angle.py

```diff
@@ -9,148 +9,148 @@
-        assert b1 == b1
-        assert b1 == b1
-        assert b1 != b2
-
-    def test_multiply(self, class_, func):
-        b1 = class_(3.14)
-        b2 = class_(2)
-        bmul = float(b1)*float(b2)
-        bmul2 = 2*float(b1)
-        assert b1*b2 == approx(bmul)
-        assert b2*b1 == approx(bmul)
-        assert 2*b1 == approx(bmul2)
-        assert b1*2 == approx(bmul2)
-
-    def test_addition(self, class_, func):
-        b1 = class_(3.14)
-        b2 = class_(2)
-        b_sum = float(b1) + float(b2)
-        assert approx(func(b_sum)) == b1+b2
-        assert approx(func(b_sum)) == float(b1)+b2
-        assert approx(func(b_sum)) == b1+float(b2)
-        assert approx(func(b_sum)) == b2+b1
-
-    def test_subtraction(self, class_, func):
-        b1 = class_(3.14)
-        b2 = class_(2)
-        b_diff = float(b1) - float(b2)
-        assert approx(func(b_diff)) == b1-b2
-        assert approx(func(b_diff)) == float(b1)-b2
-        assert approx(func(b_diff)) == b1-float(b2)
-        assert approx(-func(b_diff)) == b2-b1
-
-    def test_division(self, class_, func):
-        b1 = class_(3.14)
-        b2 = class_(2)
-        b_div = float(b1)/float(b2)
-        assert approx(b_div) == b1/b2
-        assert approx(float(b1)/2) == b1/2
-        assert approx(b_div) == float(b1)/b2
-
-    def test_comparison(self, class_, func):
-        b1 = class_(deg2rad(30))
-        b2 = class_(deg2rad(10))
-        assert b1 == b1
-        assert b1 > b2
-        assert b1 >= b1
-        assert b1 <= b1
-        assert b2 < b1
-
-    def test_trig(self, class_, func):
-        b1 = class_(deg2rad(22.0))
-        b2 = float(b1)
-
-        assert np.sin(b1) == np.sin(b2)
-        assert np.cos(b1) == np.cos(b2)
-        assert np.tan(b1) == np.tan(b2)
-        assert np.sinh(b1) == np.sinh(b2)
-        assert np.cosh(b1) == np.cosh(b2)
-        assert np.tanh(b1) == np.tanh(b2)
-        assert np.rad2deg(b1) == np.rad2deg(b2)
-
-    def test_misc(self, class_, func):
-        b1 = class_(deg2rad(22.0))
-        b2 = float(b1)
-
-        assert ceil(b1) == ceil(b2)
-        assert floor(b1) == floor(b2)
-        assert trunc(b1) == trunc(b2)
-        assert round(b1, 2) == round(b2, 2)
-        assert b1 % 0.1 == b2 % 0.1
-        assert +b1 == +b2
-        assert approx(-b1) == -b2
-
-    def test_degrees(self, class_, func):
-        b1 = class_(np.pi/4)  # pi/4 radians = 45 degrees
-        assert b1.degrees == 45.0
-
-    def test_average(self, class_, func):
-        val = np.pi/4
+@fixture(params=[Bearing, Elevation, Longitude, Latitude])
+def class_(request):
+    return request.param
+
+
+@fixture()
+def func(class_):
+    return mod_bearing if issubclass(class_, Bearing) else mod_elevation
+
+
+def test_bearing_init(class_, func):
+    b1 = class_(3.14)
+    b2 = class_(deg2rad(700))
+    assert approx(b1) == func(3.14)
+    assert approx(b2) == func(deg2rad(-20))
+    assert b1 == b1
+    assert b1 == b1
```

Comment:
Is this mean to test the same thing twice?
```suggestion
    assert b1 == b1
    assert b2 == b2
```